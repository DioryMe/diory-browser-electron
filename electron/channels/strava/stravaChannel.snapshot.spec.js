const { stravaChannel } = require('./stravaChannel')

describe('stravaChannel', () => {
  it('generates diorys from strava activity.csv and activity gps files', async () => {
    const { diorys } = await stravaChannel('some-event', {
      filePath: __dirname + '/example/activities.csv',
    })

    expect(diorys).toMatchSnapshot()
  })
})
