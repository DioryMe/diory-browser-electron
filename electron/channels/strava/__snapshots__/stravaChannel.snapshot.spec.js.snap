// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`stravaChannel generates diorys from strava activity.csv and activity gps files 1`] = `
Array [
  Object {
    "data": Object {
      "@context": "https://schema.org",
      "@type": "ExerciseAction",
      "distance": "0.4 km",
      "duration": "00:01:37",
      "endTime": "2015-01-20T16:09:21.000Z",
      "exerciseType": "skating",
      "startTime": "2015-01-20T16:07:44.000Z",
    },
    "date": "2015-01-20T16:07:44.000Z",
    "id": "strava_243698323",
    "latitude": 61.495461,
    "longitude": 23.744284,
    "text": "Evening Activity, Ice Skate",
  },
  Object {
    "data": Object {
      "@context": "https://schema.org",
      "@type": "ExerciseAction",
      "distance": "14 km",
      "duration": "00:43:57",
      "endTime": "2015-01-21T15:21:57.000Z",
      "exerciseType": "skiing",
      "startTime": "2015-01-21T14:17:04.000Z",
    },
    "date": "2015-01-21T14:17:04.000Z",
    "id": "strava_244109093",
    "latitude": 61.457613,
    "longitude": 23.665794,
    "text": "Evening Activity, Nordic Ski",
  },
  Object {
    "data": Object {
      "@context": "https://schema.org",
      "@type": "ExerciseAction",
      "distance": "4.8 km",
      "duration": "00:22:59",
      "endTime": "2015-03-09T05:53:24.000Z",
      "exerciseType": "running",
      "startTime": "2015-03-09T05:26:51.000Z",
    },
    "date": "2015-03-09T05:26:51.000Z",
    "id": "strava_265362664",
    "latitude": 61.47023,
    "longitude": 23.873692,
    "text": "Morning Run",
  },
  Object {
    "data": Object {
      "@context": "https://schema.org",
      "@type": "ExerciseAction",
      "distance": "16.9 km",
      "duration": "01:00:02",
      "endTime": "2015-03-11T04:31:39.000Z",
      "exerciseType": "cycling",
      "startTime": "2015-03-11T03:24:32.000Z",
    },
    "date": "2015-03-11T03:24:32.000Z",
    "id": "strava_266337693",
    "latitude": 61.477465,
    "longitude": 23.875392,
    "text": "Morning Ride",
  },
  Object {
    "data": Object {
      "@context": "https://schema.org",
      "@type": "ExerciseAction",
      "distance": "17.1 km",
      "duration": "00:55:56",
      "endTime": "2015-03-11T11:04:57.000Z",
      "exerciseType": "cycling",
      "startTime": "2015-03-11T09:07:38.000Z",
    },
    "date": "2015-03-11T09:07:38.000Z",
    "id": "strava_266451798",
    "latitude": 61.476394,
    "longitude": 23.879225,
    "text": "Lunch Ride",
  },
  Object {
    "data": Object {
      "@context": "https://schema.org",
      "@type": "ExerciseAction",
      "distance": "20.9 km",
      "duration": "01:35:30",
      "endTime": "2015-03-12T06:00:08.000Z",
      "exerciseType": "running",
      "startTime": "2015-03-12T03:57:19.000Z",
    },
    "date": "2015-03-12T03:57:19.000Z",
    "id": "strava_266863490",
    "latitude": 61.471984,
    "longitude": 23.874904,
    "text": "Morning Run",
  },
  Object {
    "data": Object {
      "@context": "https://schema.org",
      "@type": "ExerciseAction",
      "distance": "8.7 km",
      "duration": "00:31:51",
      "endTime": "2015-03-13T13:58:31.000Z",
      "exerciseType": "skiing",
      "startTime": "2015-03-13T13:24:31.000Z",
    },
    "date": "2015-03-13T13:24:31.000Z",
    "id": "strava_267498382",
    "latitude": 61.511024,
    "longitude": 23.806557,
    "text": "Afternoon Ski, Nordic Ski",
  },
  Object {
    "data": Object {
      "@context": "https://schema.org",
      "@type": "ExerciseAction",
      "distance": "11.3 km",
      "duration": "00:46:56",
      "endTime": "2015-03-19T16:11:30.000Z",
      "exerciseType": "running",
      "startTime": "2015-03-19T15:21:56.000Z",
    },
    "date": "2015-03-19T15:21:56.000Z",
    "id": "strava_270792369",
    "latitude": 61.509474,
    "longitude": 23.807707,
    "text": "Evening Run",
  },
  Object {
    "data": Object {
      "@context": "https://schema.org",
      "@type": "ExerciseAction",
      "distance": "5.5 km",
      "duration": "00:29:51",
      "endTime": "2015-03-22T16:53:24.000Z",
      "exerciseType": "running",
      "startTime": "2015-03-22T16:22:40.000Z",
    },
    "date": "2015-03-22T16:22:40.000Z",
    "id": "strava_272498540",
    "latitude": 61.475518,
    "longitude": 23.881178,
    "text": "Evening Run",
  },
  Object {
    "data": Object {
      "@context": "https://schema.org",
      "@type": "ExerciseAction",
      "distance": "43.2 km",
      "duration": "02:15:15",
      "endTime": "2015-03-21T09:18:04.000Z",
      "exerciseType": "skiing",
      "heartrate": 170,
      "startTime": "2015-03-21T07:00:59.000Z",
    },
    "date": "2015-03-21T07:00:59.000Z",
    "id": "strava_272859333",
    "latitude": undefined,
    "longitude": undefined,
    "text": "03/21/2015 Finland, Nordic Ski",
  },
  Object {
    "data": Object {
      "@context": "https://schema.org",
      "@type": "ExerciseAction",
      "distance": "0 km",
      "duration": "00:49:30",
      "endTime": "2015-01-22T17:31:22.000Z",
      "exerciseType": "skiing",
      "startTime": "2015-01-22T16:41:52.000Z",
    },
    "date": "2015-01-22T16:41:52.000Z",
    "id": "strava_273325662",
    "latitude": undefined,
    "longitude": undefined,
    "text": "01/22/2015 Finland, Nordic Ski",
  },
  Object {
    "data": Object {
      "@context": "https://schema.org",
      "@type": "ExerciseAction",
      "distance": "0 km",
      "duration": "02:50:45",
      "endTime": "2015-01-17T12:05:47.000Z",
      "exerciseType": "skiing",
      "startTime": "2015-01-17T09:15:02.000Z",
    },
    "date": "2015-01-17T09:15:02.000Z",
    "id": "strava_273325663",
    "latitude": undefined,
    "longitude": undefined,
    "text": "01/17/2015 Finland, Nordic Ski",
  },
  Object {
    "data": Object {
      "@context": "https://schema.org",
      "@type": "ExerciseAction",
      "distance": "0 km",
      "duration": "01:10:40",
      "endTime": "2015-01-27T15:26:48.000Z",
      "exerciseType": "running",
      "startTime": "2015-01-27T14:16:08.000Z",
    },
    "date": "2015-01-27T14:16:08.000Z",
    "id": "strava_273325667",
    "latitude": undefined,
    "longitude": undefined,
    "text": "01/27/2015 Finland, Run",
  },
  Object {
    "data": Object {
      "@context": "https://schema.org",
      "@type": "ExerciseAction",
      "distance": "0 km",
      "duration": "01:04:00",
      "endTime": "2015-01-27T17:21:46.000Z",
      "exerciseType": "running",
      "startTime": "2015-01-27T16:17:46.000Z",
    },
    "date": "2015-01-27T16:17:46.000Z",
    "id": "strava_273325674",
    "latitude": undefined,
    "longitude": undefined,
    "text": "01/27/2015 Finland, Run",
  },
  Object {
    "data": Object {
      "@context": "https://schema.org",
      "@type": "ExerciseAction",
      "distance": "0 km",
      "duration": "00:48:05",
      "endTime": "2015-01-31T12:51:50.000Z",
      "exerciseType": "running",
      "startTime": "2015-01-31T12:03:45.000Z",
    },
    "date": "2015-01-31T12:03:45.000Z",
    "id": "strava_273325675",
    "latitude": undefined,
    "longitude": undefined,
    "text": "01/31/2015 Finland, Run",
  },
  Object {
    "data": Object {
      "@context": "https://schema.org",
      "@type": "ExerciseAction",
      "distance": "0 km",
      "duration": "00:29:55",
      "endTime": "2015-02-01T15:21:04.000Z",
      "exerciseType": "skiing",
      "startTime": "2015-02-01T14:51:09.000Z",
    },
    "date": "2015-02-01T14:51:09.000Z",
    "id": "strava_273325677",
    "latitude": undefined,
    "longitude": undefined,
    "text": "02/01/2015 Finland, Nordic Ski",
  },
  Object {
    "data": Object {
      "@context": "https://schema.org",
      "@type": "ExerciseAction",
      "distance": "0 km",
      "duration": "00:29:15",
      "endTime": "2015-02-03T16:20:03.000Z",
      "exerciseType": "skiing",
      "startTime": "2015-02-03T15:50:48.000Z",
    },
    "date": "2015-02-03T15:50:48.000Z",
    "id": "strava_273325683",
    "latitude": undefined,
    "longitude": undefined,
    "text": "02/03/2015 Finland, Nordic Ski",
  },
  Object {
    "data": Object {
      "@context": "https://schema.org",
      "@type": "ExerciseAction",
      "distance": "0 km",
      "duration": "00:39:20",
      "endTime": "2015-02-07T07:28:11.000Z",
      "exerciseType": "skiing",
      "startTime": "2015-02-07T06:48:51.000Z",
    },
    "date": "2015-02-07T06:48:51.000Z",
    "id": "strava_273325687",
    "latitude": undefined,
    "longitude": undefined,
    "text": "02/07/2015 Finland, Nordic Ski",
  },
  Object {
    "data": Object {
      "@context": "https://schema.org",
      "@type": "ExerciseAction",
      "distance": "0 km",
      "duration": "00:04:37",
      "endTime": "2015-02-14T06:19:05.000Z",
      "exerciseType": "skiing",
      "startTime": "2015-02-14T06:14:28.000Z",
    },
    "date": "2015-02-14T06:14:28.000Z",
    "id": "strava_273325689",
    "latitude": undefined,
    "longitude": undefined,
    "text": "02/14/2015 Finland, Nordic Ski",
  },
  Object {
    "data": Object {
      "@context": "https://schema.org",
      "@type": "ExerciseAction",
      "distance": "0 km",
      "duration": "00:49:30",
      "endTime": "2015-03-14T13:52:08.000Z",
      "exerciseType": "skiing",
      "startTime": "2015-03-14T13:02:38.000Z",
    },
    "date": "2015-03-14T13:02:38.000Z",
    "id": "strava_273325690",
    "latitude": undefined,
    "longitude": undefined,
    "text": "03/14/2015 Finland, Nordic Ski",
  },
  Object {
    "data": Object {
      "@context": "https://schema.org",
      "@type": "ExerciseAction",
      "distance": "0 km",
      "duration": "00:57:50",
      "endTime": "2015-03-19T16:20:20.000Z",
      "exerciseType": "skiing",
      "heartrate": 136,
      "startTime": "2015-03-19T15:22:30.000Z",
    },
    "date": "2015-03-19T15:22:30.000Z",
    "id": "strava_273325692",
    "latitude": undefined,
    "longitude": undefined,
    "text": "03/19/2015 Finland, Nordic Ski",
  },
  Object {
    "data": Object {
      "@context": "https://schema.org",
      "@type": "ExerciseAction",
      "distance": "0 km",
      "duration": "04:12:25",
      "endTime": "2015-02-21T08:12:50.000Z",
      "exerciseType": "skiing",
      "heartrate": 150,
      "startTime": "2015-02-21T04:00:25.000Z",
    },
    "date": "2015-02-21T04:00:25.000Z",
    "id": "strava_273325700",
    "latitude": undefined,
    "longitude": undefined,
    "text": "02/20/2015 Finland, Nordic Ski",
  },
  Object {
    "data": Object {
      "@context": "https://schema.org",
      "@type": "ExerciseAction",
      "distance": "5.3 km",
      "duration": "00:30:19",
      "endTime": "2015-03-26T06:12:47.000Z",
      "exerciseType": "running",
      "startTime": "2015-03-26T05:33:42.000Z",
    },
    "date": "2015-03-26T05:33:42.000Z",
    "id": "strava_274251044",
    "latitude": 61.476282,
    "longitude": 23.879709,
    "text": "Morning Run",
  },
  Object {
    "data": Object {
      "@context": "https://schema.org",
      "@type": "ExerciseAction",
      "distance": "12.4 km",
      "duration": "01:07:55",
      "endTime": "2015-03-29T08:08:22.000Z",
      "exerciseType": "running",
      "startTime": "2015-03-29T06:52:29.000Z",
    },
    "date": "2015-03-29T06:52:29.000Z",
    "id": "strava_275786550",
    "latitude": 61.475283,
    "longitude": 23.878303,
    "text": "Lunch Run",
  },
  Object {
    "data": Object {
      "@context": "https://schema.org",
      "@type": "ExerciseAction",
      "distance": "11.7 km",
      "duration": "01:04:16",
      "endTime": "2015-03-31T04:20:27.000Z",
      "exerciseType": "running",
      "startTime": "2015-03-31T03:05:56.000Z",
    },
    "date": "2015-03-31T03:05:56.000Z",
    "id": "strava_276780534",
    "latitude": 61.476848,
    "longitude": 23.878149,
    "text": "Morning Run",
  },
  Object {
    "data": Object {
      "@context": "https://schema.org",
      "@type": "ExerciseAction",
      "distance": "0.1 km",
      "duration": "00:49:31",
      "endTime": "2015-04-02T15:07:14.000Z",
      "exerciseType": "swimming",
      "startTime": "2015-04-02T14:17:43.000Z",
    },
    "date": "2015-04-02T14:17:43.000Z",
    "id": "strava_278058793",
    "latitude": 61.498476,
    "longitude": 23.806199,
    "text": "Evening Swim",
  },
  Object {
    "data": Object {
      "@context": "https://schema.org",
      "@type": "ExerciseAction",
      "distance": "9.6 km",
      "duration": "00:38:41",
      "endTime": "2015-04-04T12:50:15.000Z",
      "exerciseType": "running",
      "startTime": "2015-04-04T12:06:19.000Z",
    },
    "date": "2015-04-04T12:06:19.000Z",
    "id": "strava_279098107",
    "latitude": 63.279424,
    "longitude": 28.034933,
    "text": "Evening Run",
  },
  Object {
    "data": Object {
      "@context": "https://schema.org",
      "@type": "ExerciseAction",
      "distance": "22.3 km",
      "duration": "02:14:45",
      "endTime": "2015-04-05T09:31:30.000Z",
      "exerciseType": "skiing",
      "startTime": "2015-04-05T04:57:39.000Z",
    },
    "date": "2015-04-05T04:57:39.000Z",
    "id": "strava_279609872",
    "latitude": 63.279532,
    "longitude": 28.033737,
    "text": "Morning Activity, Nordic Ski",
  },
  Object {
    "data": Object {
      "@context": "https://schema.org",
      "@type": "ExerciseAction",
      "distance": "9.7 km",
      "duration": "01:01:08",
      "endTime": "2015-04-07T08:11:37.000Z",
      "exerciseType": "running",
      "startTime": "2015-04-07T05:47:08.000Z",
    },
    "date": "2015-04-07T05:47:08.000Z",
    "id": "strava_280799693",
    "latitude": 63.279848,
    "longitude": 28.034503,
    "text": "Lunch Run",
  },
  Object {
    "data": Object {
      "@context": "https://schema.org",
      "@type": "ExerciseAction",
      "distance": "4.6 km",
      "duration": "00:20:21",
      "endTime": "2015-04-08T12:47:15.000Z",
      "exerciseType": "skiing",
      "startTime": "2015-04-08T12:24:21.000Z",
    },
    "date": "2015-04-08T12:24:21.000Z",
    "id": "strava_281516081",
    "latitude": 63.279572,
    "longitude": 28.034375,
    "text": "Evening Activity, Backcountry Ski",
  },
  Object {
    "data": Object {
      "@context": "https://schema.org",
      "@type": "ExerciseAction",
      "distance": "4.6 km",
      "duration": "00:23:17",
      "endTime": "2015-04-08T13:17:34.000Z",
      "exerciseType": "running",
      "startTime": "2015-04-08T12:51:02.000Z",
    },
    "date": "2015-04-08T12:51:02.000Z",
    "id": "strava_281532322",
    "latitude": 63.27886,
    "longitude": 28.034671,
    "text": "Evening Run",
  },
  Object {
    "data": Object {
      "@context": "https://schema.org",
      "@type": "ExerciseAction",
      "distance": "0.1 km",
      "duration": "01:29:31",
      "endTime": "2015-04-09T15:49:28.000Z",
      "exerciseType": "swimming",
      "startTime": "2015-04-09T14:19:57.000Z",
    },
    "date": "2015-04-09T14:19:57.000Z",
    "id": "strava_282249210",
    "latitude": 61.498657,
    "longitude": 23.806046,
    "text": "Evening Swim",
  },
  Object {
    "data": Object {
      "@context": "https://schema.org",
      "@type": "ExerciseAction",
      "distance": "16 km",
      "duration": "01:06:56",
      "endTime": "2015-04-15T12:05:05.000Z",
      "exerciseType": "running",
      "startTime": "2015-04-15T09:51:05.000Z",
    },
    "date": "2015-04-15T09:51:05.000Z",
    "id": "strava_285897739",
    "latitude": 61.477182,
    "longitude": 23.877791,
    "text": "Afternoon Run",
  },
  Object {
    "data": Object {
      "@context": "https://schema.org",
      "@type": "ExerciseAction",
      "distance": "26.7 km",
      "duration": "02:24:20",
      "endTime": "2015-04-17T07:50:26.000Z",
      "exerciseType": "cycling",
      "startTime": "2015-04-17T05:26:06.000Z",
    },
    "date": "2015-04-17T05:26:06.000Z",
    "id": "strava_287048880",
    "latitude": 61.476993,
    "longitude": 23.874539,
    "text": "Lunch Ride",
  },
  Object {
    "data": Object {
      "@context": "https://schema.org",
      "@type": "ExerciseAction",
      "distance": "19.8 km",
      "duration": "01:05:45",
      "endTime": "2015-04-25T06:23:39.000Z",
      "exerciseType": "cycling",
      "startTime": "2015-04-25T04:50:40.000Z",
    },
    "date": "2015-04-25T04:50:40.000Z",
    "id": "strava_292105660",
    "latitude": 61.158313,
    "longitude": 23.886339,
    "text": "Morning Run, Ride",
  },
  Object {
    "data": Object {
      "@context": "https://schema.org",
      "@type": "ExerciseAction",
      "distance": "22.5 km",
      "duration": "02:05:56",
      "endTime": "2015-04-28T04:27:37.000Z",
      "exerciseType": "running",
      "startTime": "2015-04-28T02:15:14.000Z",
    },
    "date": "2015-04-28T02:15:14.000Z",
    "id": "strava_293994108",
    "latitude": 61.476126,
    "longitude": 23.879472,
    "text": "Morning Run",
  },
  Object {
    "data": Object {
      "@context": "https://schema.org",
      "@type": "ExerciseAction",
      "distance": "14.1 km",
      "duration": "00:42:11",
      "endTime": "2015-10-19T13:24:24.000Z",
      "exerciseType": "running",
      "startTime": "2015-10-19T11:54:27.000Z",
    },
    "date": "2015-10-19T11:54:27.000Z",
    "id": "strava_416199595",
    "latitude": 61.491134,
    "longitude": 23.758293,
    "text": "Afternoon Run",
  },
  Object {
    "data": Object {
      "@context": "https://schema.org",
      "@type": "ExerciseAction",
      "distance": "6.8 km",
      "duration": "00:17:54",
      "endTime": "2015-10-20T12:50:24.000Z",
      "exerciseType": "running",
      "startTime": "2015-10-20T12:07:53.000Z",
    },
    "date": "2015-10-20T12:07:53.000Z",
    "id": "strava_416798956",
    "latitude": 61.474682,
    "longitude": 23.885622,
    "text": "Evening Run",
  },
  Object {
    "data": Object {
      "@context": "https://schema.org",
      "@type": "ExerciseAction",
      "distance": "25.5 km",
      "duration": "01:06:54",
      "endTime": "2015-11-27T17:32:38.000Z",
      "exerciseType": "skating",
      "startTime": "2015-11-27T16:25:44.000Z",
    },
    "date": "2015-11-27T16:25:44.000Z",
    "id": "strava_440370309",
    "latitude": 60.234719,
    "longitude": 24.959314,
    "text": "Skate, Ice Skate",
  },
]
`;

exports[`stravaChannel generates diorys from strava activity.csv file 1`] = `
Array [
  Object {
    "data": Object {
      "@context": "https://schema.org",
      "@type": "ExerciseAction",
      "distance": "0.4 km",
      "duration": "00:01:37",
      "endTime": "2015-01-20T16:09:21.000Z",
      "exerciseType": "skating",
      "startTime": "2015-01-20T16:07:44.000Z",
    },
    "date": "2015-01-20T16:07:44.000Z",
    "id": "strava_243698323",
    "latitude": 61.495461,
    "longitude": 23.744284,
    "text": "Evening Activity, Ice Skate",
  },
  Object {
    "data": Object {
      "@context": "https://schema.org",
      "@type": "ExerciseAction",
      "distance": "14 km",
      "duration": "00:43:57",
      "endTime": "2015-01-21T15:21:57.000Z",
      "exerciseType": "skiing",
      "startTime": "2015-01-21T14:17:04.000Z",
    },
    "date": "2015-01-21T14:17:04.000Z",
    "id": "strava_244109093",
    "latitude": 61.457613,
    "longitude": 23.665794,
    "text": "Evening Activity, Nordic Ski",
  },
  Object {
    "data": Object {
      "@context": "https://schema.org",
      "@type": "ExerciseAction",
      "distance": "4.8 km",
      "duration": "00:22:59",
      "endTime": "2015-03-09T05:53:24.000Z",
      "exerciseType": "running",
      "startTime": "2015-03-09T05:26:51.000Z",
    },
    "date": "2015-03-09T05:26:51.000Z",
    "id": "strava_265362664",
    "latitude": 61.47023,
    "longitude": 23.873692,
    "text": "Morning Run",
  },
  Object {
    "data": Object {
      "@context": "https://schema.org",
      "@type": "ExerciseAction",
      "distance": "16.9 km",
      "duration": "01:00:02",
      "endTime": "2015-03-11T04:31:39.000Z",
      "exerciseType": "cycling",
      "startTime": "2015-03-11T03:24:32.000Z",
    },
    "date": "2015-03-11T03:24:32.000Z",
    "id": "strava_266337693",
    "latitude": 61.477465,
    "longitude": 23.875392,
    "text": "Morning Ride",
  },
  Object {
    "data": Object {
      "@context": "https://schema.org",
      "@type": "ExerciseAction",
      "distance": "17.1 km",
      "duration": "00:55:56",
      "endTime": "2015-03-11T11:04:57.000Z",
      "exerciseType": "cycling",
      "startTime": "2015-03-11T09:07:38.000Z",
    },
    "date": "2015-03-11T09:07:38.000Z",
    "id": "strava_266451798",
    "latitude": 61.476394,
    "longitude": 23.879225,
    "text": "Lunch Ride",
  },
  Object {
    "data": Object {
      "@context": "https://schema.org",
      "@type": "ExerciseAction",
      "distance": "20.9 km",
      "duration": "01:35:30",
      "endTime": "2015-03-12T06:00:08.000Z",
      "exerciseType": "running",
      "startTime": "2015-03-12T03:57:19.000Z",
    },
    "date": "2015-03-12T03:57:19.000Z",
    "id": "strava_266863490",
    "latitude": 61.471984,
    "longitude": 23.874904,
    "text": "Morning Run",
  },
  Object {
    "data": Object {
      "@context": "https://schema.org",
      "@type": "ExerciseAction",
      "distance": "8.7 km",
      "duration": "00:31:51",
      "endTime": "2015-03-13T13:58:31.000Z",
      "exerciseType": "skiing",
      "startTime": "2015-03-13T13:24:31.000Z",
    },
    "date": "2015-03-13T13:24:31.000Z",
    "id": "strava_267498382",
    "latitude": 61.511024,
    "longitude": 23.806557,
    "text": "Afternoon Ski, Nordic Ski",
  },
  Object {
    "data": Object {
      "@context": "https://schema.org",
      "@type": "ExerciseAction",
      "distance": "11.3 km",
      "duration": "00:46:56",
      "endTime": "2015-03-19T16:11:30.000Z",
      "exerciseType": "running",
      "startTime": "2015-03-19T15:21:56.000Z",
    },
    "date": "2015-03-19T15:21:56.000Z",
    "id": "strava_270792369",
    "latitude": 61.509474,
    "longitude": 23.807707,
    "text": "Evening Run",
  },
  Object {
    "data": Object {
      "@context": "https://schema.org",
      "@type": "ExerciseAction",
      "distance": "5.5 km",
      "duration": "00:29:51",
      "endTime": "2015-03-22T16:53:24.000Z",
      "exerciseType": "running",
      "startTime": "2015-03-22T16:22:40.000Z",
    },
    "date": "2015-03-22T16:22:40.000Z",
    "id": "strava_272498540",
    "latitude": 61.475518,
    "longitude": 23.881178,
    "text": "Evening Run",
  },
  Object {
    "data": Object {
      "@context": "https://schema.org",
      "@type": "ExerciseAction",
      "distance": "43.2 km",
      "duration": "02:15:15",
      "endTime": "2015-03-21T09:18:04.000Z",
      "exerciseType": "skiing",
      "heartrate": 170,
      "startTime": "2015-03-21T07:00:59.000Z",
    },
    "date": "2015-03-21T07:00:59.000Z",
    "id": "strava_272859333",
    "latitude": undefined,
    "longitude": undefined,
    "text": "03/21/2015 Finland, Nordic Ski",
  },
]
`;
